File: Numbers/program25.py
'''

    File name : program25.py
    Description : Accept number from user and display there factorial
    Author : Shrishti Sunil Gaikwad
    Date : 11/05/2025

'''

def Factorial(iNo):
    iFact = 1

    for i in range(1,iNo+1):
        iFact = iFact * i

    return iFact

def main():
    print("Enter number : ")
    iValue = int(input())

    iRet = Factorial(iValue)

    print("Factorial is : ",iRet)

if __name__ == "__main__":
    main()

File: Numbers/program25.c
//////////////////////////////////////////////
//
//  File name : program25.c
//  Description : Check whether it is factorial or not
//  Author : Shrishti Sunil Gaikwad
//  Date : 11/05/2025
//
////////////////////////////////////////////////

#include <stdio.h>
#include<stdbool.h>

//without using flag
bool CheckPrime(int iNo)
{
   int iCnt = 0;
   bool bFlag = true;

    if(iNo < 0)
    {
        iNo = -iNo;
    }

    for(iCnt = 2;(iCnt <= (iNo/2)); iCnt++)
    {
        if((iNo % iCnt) == 0)
        {
    
            break;
        }
    }

    return (iCnt > (iNo/2));
}

int main()
{
    int iValue = 0;
    bool bRet = false;

    printf("Enter number:\n");
    scanf("%d",&iValue);

    bRet = CheckPrime(iValue);

    if(bRet == true)
    {
        printf("%d is a Prime number",iValue);
    }
    else
    {
        printf("%d is Not Prime number",iValue);
    }
    return 0;
}